Sources/PromotedCore/NetworkConnection.swift

```diff
@@ -61,8 +61,6 @@ public extension NetworkConnection {
       return try message.serializedData()
     case .json:
       return try message.jsonUTF8Data()
-    case .unknown:
```

Comment:
What happens here now?  Does it throw?  This is more me learning about this.

Reply:
That's a good question. Although the normal Swift compiler and runtime don't allow you to create any other values for the enum, it's possible to create invalid values using key-value coding. I ran the following experiment:
```swift
@objc class BadEnumTest: NSObject {
  @objc enum E: Int {
    case foo = 0
    case bar = 1
  }
  @objc var e: E

  @objc override init() {
    e = .foo
  }
}

func hello(_ e: BadEnumTest.E) -> Int {
  switch e {
  case .foo:
    return 10
  case .bar:
    return 20
  }
}

let t = BadEnumTest()
t.setValue("invalid", forKey: "e")
let f = hello(t.e)
print("\(f)")
```
In the last block, `t.e` has an invalid value which resolves to `0`. If the enum has a case `0` then `t.e` resolves to that value for the enum (above it's the `.foo` case).

However, if you don't have a case `0`, say the following:
```swift
@objc enum E: Int {
  case foo = 1
  case bar = 2
}
```
Then running the same block of code to generate an invalid enum results in a crash (`Fatal error: unexpected enum case 'E(rawValue: 0)'`) when evaluating `t.e`!

The lesson from this exercise is that `Int` enum values should always have a case `0`, and preferably that case should be a default or harmless value.

Thanks for pointing this out! I'm pushing another commit to fix this in this PR.